src/Config.php

```diff
@@ -437,9 +440,16 @@ public function onProduction() : bool
             return false;
         }
 
-        $prodBranch = $this->onDedicated() ? 'production' : 'master';
-
-        return $this->getValue('BRANCH') == $prodBranch;
+        // we need to first confirm that we actually have the `ENVIRONMENT_TYPE` variable, 
+        // because not all legacy containers will have this
+        if($this->hasVariable('ENVIRONMENT_TYPE')) {
+            // correct way of checking production branch
+            return $this->getValue('ENVIRONMENT_TYPE') == 'production';
+        } else {
+            // legacy way of checking production type
+            $prodBranch = $this->onDedicated() ? 'production' : 'master';
```

Comment:
`master` ?

Reply:
well yeah, that's the backwards compatibility. Ideally we won't ever get into this else statement. But if we do, the functionality needs to be the same.